src/util/notifications_util.js

```diff
@@ -494,10 +499,31 @@ function get_notification_logger(locking, namespace, poll_interval) {
         namespace = node_name + '_' + config.NOTIFICATION_LOG_NS;
     }
 
-    return new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
+    const logger = new PersistentLogger(config.NOTIFICATION_LOG_DIR, namespace, {
         locking,
         poll_interval,
     });
+
+    //initialize writes_counter, used in check_free_space
+    logger.writes_counter = 0;
+
+    return logger;
+}
+
+//If space check is configures, create an event in case free space is below threshold.
+function check_free_space(req) {
+    if (!req.object_sdk.nsfs_config_root || !config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        return;
+    }
+    req.notification_logger.writes_counter += 1;
+
+    if (req.notification_logger.writes_counter > config.NOTIFICATION_REQ_PER_SPACE_CHECK) {
+        req.notification_logger.writes_counter = 0;
+        const fs_stat = fs.statfsSync(config.NOTIFICATION_LOG_DIR);
+        if (fs_stat.bavail / fs_stat.blocks < config.NOTIFICATION_SPACE_CHECK_THRESHOLD) {
+            new NoobaaEvent(NoobaaEvent.NOTIFICATION_LOW_SPACE).create_event(null, {fs_stat});
```

Comment:
Okay, this we need to think...

Once we reach the space limit where we have to raise an event, means we are in danger of running out of space?
If so, this write onward, we should raise this event after every write and not after "NOTIFICATION_REQ_PER_SPACE_CHECK"

so I think we sould reset req.notification_logger.writes_counter = 0; 
if we are okay with the space and 

if (fs_stat.bavail / fs_stat.blocks < config.NOTIFICATION_SPACE_CHECK_THRESHOLD)

is not true.

Reply:
I'm not sure, sounds excessive logging for a situation that is not necessarily wrong or avoidable, where functionality is not necessarily degrade.
I would guess that if users ignore the warn event writing a lot more warn event will probably won't help.